packages/components/src/color-picker/hex-input.tsx

```diff
@@ -24,6 +24,10 @@ interface HexInputProps {
 }
 
 export const HexInput = ( { color, onChange, enableAlpha }: HexInputProps ) => {
+	const handleChange = ( nextValue ) => {
+		const hexValue = ( '#' + nextValue ).replace( '##', '#' );
+		onChange( colord( hexValue ) );
+	};
 	const handleValidate = ( value: string ) => {
```

Comment:
I noticed this while reviewing that we never call the validation function. ColorHexInputControl needs the `isPressEnterToChange` property to check values.

Here's what the behavior looks like when this is set. If invalid values are entered, we keep the last color value. The tradeoff though is that the color value needs an <kbd>enter</kbd> keypress, versus updating as we type.

https://user-images.githubusercontent.com/1270189/153684366-5a83252c-4f6c-4d71-adb3-73abf7e4263a.mp4

Reply:
Oh so I don't think we need to change the behavior to add `isPressEnterToChange`, I was just noting that the `handleValidate` doesn't do anything. Unless other folks chime in, I think it's okay to remove the `handleValidate` handler, so folks don't wonder why nothing is happening. https://github.com/WordPress/gutenberg/blob/ea32685fbabf070e022112b0b94c9e4cfa8ff307/packages/components/src/input-control/input-field.tsx#L160-L162

> Are you able to actually trigger the validation handler? 

Yes, I'm able to. One thing I am running into is that the max length set on the input will truncate the values we are interested in. This appears to be the case for osx / Chrome,FF,Safari combos. Which os/browser are you using? I wonder if we're seeing some subtle browser differences.

https://user-images.githubusercontent.com/1270189/153907935-9427ed6e-ff73-4234-add2-2f78ec1f81e2.mp4